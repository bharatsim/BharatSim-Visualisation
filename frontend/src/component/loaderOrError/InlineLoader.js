import React from 'react';
import PropTypes from 'prop-types';

import { FormHelperText } from '@material-ui/core';

import { loaderStates } from '../../hook/useLoader';

function InlineLoader({ status, message }) {
  return (
    <>
      {!!status && <FormHelperText error={status === loaderStates.ERROR}>{message}</FormHelperText>}
    </>
  );
}

InlineLoader.propTypes = {
  status: PropTypes.oneOf(['', ...Object.values(loaderStates)]).isRequired,
  message: PropTypes.string.isRequired,
};

export default InlineLoader;
