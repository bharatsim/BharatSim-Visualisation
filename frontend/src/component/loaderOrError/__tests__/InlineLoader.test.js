import React from 'react';

import { render } from '@testing-library/react';
import InlineLoader from '../InlineLoader';

describe('<InlineLoader />', () => {
  it('should match snapshot', () => {
    const { container } = render(<InlineLoader status="LOADING" message="Loader..." />);

    expect(container).toMatchSnapshot();
  });
});
