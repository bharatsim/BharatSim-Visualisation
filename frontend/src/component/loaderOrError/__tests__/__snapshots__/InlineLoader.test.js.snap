// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineLoader /> should match snapshot 1`] = `
<div>
  <p
    class="MuiFormHelperText-root"
  >
    Loader...
  </p>
</div>
`;
